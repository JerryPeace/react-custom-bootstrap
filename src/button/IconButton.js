import PropTypes from 'prop-types';
import React from 'react';
import bootstrapUtils from '../utils/bootstrapUtils';
import classNames from 'classnames';
import styleMaps from '../styleMaps';
import Glyphicon from '../glyphicon/Glyphicon';

class IconButton extends React.Component {

    static propTypes = {
        bsSize: PropTypes.oneOf(['lg', 'sm', 'xs']),
        type: PropTypes.oneOf(['button', 'submit', 'reset']),
        block: PropTypes.bool,
        active: PropTypes.bool,
        onClick: PropTypes.func,
        glyphiconType: PropTypes.string,
        imgIconSrc: PropTypes.string,
        buttonText: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
        isDropdown: PropTypes.bool,
        isCaretStyle: PropTypes.bool,
        isIconOnly: PropTypes.bool,
        bsClass: PropTypes.string,
        bsStyle: PropTypes.string,
        standalone: PropTypes.bool
    };

    static defaultProps = {
        bsClass: 'button',
        bsStyle: 'default',
        type: 'button',
        block: false,
        active: false,
        isDropdown: false,
        isCaretStyle: false
    };

    componentDidMount() {
        console.warn('<IconButton> will be deprecated. please use <Button> instead');
    }

    renderIcon() {
        if (this.props.imgIconSrc) {
            return (<img src={this.props.imgIconSrc}/>);
        } else if (this.props.glyphiconType) {
            return (<Glyphicon bsStyle={this.props.glyphiconType} />);
        } else {
            return null;
        }
    }

    render() {
        const {
            children,
            className,
            block,
            active,
            isDropdown,
            isCaretStyle,
            bsClass,
            bsStyle,
            bsSize,
            glyphiconType,
            buttonText,
            standalone,
            isIconOnly,
            ...rest
        } = this.props;

        const {
            caret,
            'dropdown-toggle': dropdownToggle,
        } = styleMaps.CLASSES;

        const classes = {
            'btn-block': block,
            'active': active,
            'btn-icon-only': isIconOnly,
            [dropdownToggle]: isDropdown
        };
        const icon = this.renderIcon();
        const bsCls = bootstrapUtils.getCls(this.props);

        return (
            <button
                {...rest}
                className={classNames(className, classes, bsCls)}
            >
                {icon}
                {icon && ' '}
                {buttonText}
                {children}
                {isCaretStyle && ' '}
                {isCaretStyle && <span className={caret} />}
            </button>
        );
    }

}

export default IconButton;
