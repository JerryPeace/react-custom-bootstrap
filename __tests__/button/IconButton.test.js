import React from 'react';
import {mount} from 'enzyme';
import IconButton from 'button/IconButton';

describe('IconButton', () => {
    let button;

    test('glyphicon', () => {
        button = mount(<IconButton glyphiconType='pen'/>);
        expect(button).toMatchSnapshot();
    });

    test('with text', () => {
        button = mount(<IconButton glyphiconType='pen'>children</IconButton>);
        expect(button).toMatchSnapshot();
    });

    test('image icon', () => {
        button = mount(<IconButton imgIconSrc='./fake.jpg'/>);
        expect(button).toMatchSnapshot();
    });

    test('empty', () => {
        button = mount(<IconButton/>);
        expect(button).toMatchSnapshot();
    });
});
