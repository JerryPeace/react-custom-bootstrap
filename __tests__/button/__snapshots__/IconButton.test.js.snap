// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton empty 1`] = `
<IconButton
  active={false}
  block={false}
  bsClass="button"
  bsStyle="default"
  isCaretStyle={false}
  isDropdown={false}
  type="button"
>
  <button
    className="btn btn-default"
    type="button"
  />
</IconButton>
`;

exports[`IconButton glyphicon 1`] = `
<IconButton
  active={false}
  block={false}
  bsClass="button"
  bsStyle="default"
  glyphiconType="pen"
  isCaretStyle={false}
  isDropdown={false}
  type="button"
>
  <button
    className="btn btn-default"
    type="button"
  >
    <Glyphicon
      ariaHidden={true}
      bsClass="glyphicon"
      bsStyle="pen"
    >
      <span
        aria-hidden={true}
        className="glyphicon pen"
      />
    </Glyphicon>
     
  </button>
</IconButton>
`;

exports[`IconButton image icon 1`] = `
<IconButton
  active={false}
  block={false}
  bsClass="button"
  bsStyle="default"
  imgIconSrc="./fake.jpg"
  isCaretStyle={false}
  isDropdown={false}
  type="button"
>
  <button
    className="btn btn-default"
    imgIconSrc="./fake.jpg"
    type="button"
  >
    <img
      src="./fake.jpg"
    />
     
  </button>
</IconButton>
`;

exports[`IconButton with text 1`] = `
<IconButton
  active={false}
  block={false}
  bsClass="button"
  bsStyle="default"
  glyphiconType="pen"
  isCaretStyle={false}
  isDropdown={false}
  type="button"
>
  <button
    className="btn btn-default"
    type="button"
  >
    <Glyphicon
      ariaHidden={true}
      bsClass="glyphicon"
      bsStyle="pen"
    >
      <span
        aria-hidden={true}
        className="glyphicon pen"
      />
    </Glyphicon>
     
    children
  </button>
</IconButton>
`;
